site/app/models/User.php

```diff
@@ -449,8 +473,50 @@ public function getDisplayFullName() {
         return $this->getDisplayedGivenName() . ' ' . $this->getDisplayedFamilyName();
     }
 
-    public function getDisplayAbbreviatedName(): string {
-        return $this->getDisplayedGivenName() . ' ' . substr($this->getDisplayedFamilyName(), 0, 1) . '.';
+    public function getDisplayAbbreviatedName(int $last_initial_format = -1): string {
+        if ($last_initial_format < 0) {
+            $last_initial_format = $this->getLastInitialFormat();
+        }
+        $last_initial = ' ';
+        $family_name = $this->getDisplayedFamilyName();
+        if ($last_initial_format === self::LAST_INITIAL_MULTI) {
+            $spaced = preg_split('/\s+/', $family_name, -1, PREG_SPLIT_NO_EMPTY);
+            foreach ($spaced as $part) {
+                $last_initial .= $part[0] . '.';
+            }
+        }
+        elseif ($last_initial_format === self::LAST_INITIAL_HYPHEN) {
+            $spaced = preg_split('/\s+/', $family_name, -1, PREG_SPLIT_NO_EMPTY);
+            foreach ($spaced as $part) {
+                $dashed = preg_split('/-/', $part, -1, PREG_SPLIT_NO_EMPTY);
```

Comment:
No need to use `prep_split` for a single character here.  You can use `explode` for this.